### DISCUSSION TITLE
The use of units

### DISCUSSION THREAD

**Discussion by sacdevops** [dotnet/maui#333] [Ideas]
Already had a conversation with Redth about it today, but would also like to hear the community's opinion on it. Would the introduction of units make your work easier? These are already known from HTML / CSS and would then be similar. Pixel units are standard and need not be mentioned. Units like %, vh and vw would be things that could still be included.
There are ways to get the length and width of the device and then apply them later, but an integrated unit would make things a lot easier in my opinion.

Two scenarios where it could clearly make sense:
1. When using XAML design. In this way, objects could be delimited directly, such as a <BoxView HeightRequest = "30vh" />. Would take 30% of the height of the device.
2. The AbsoluteLayout with a connection for the calculation of data could become much more powerful. An example for this <BoxView AbsoluteLayout.SetBounds = "50, 20%, 100, 30% - 40" /> That would both remove the flags and allow you to make perfect adjustments to objects.

I'm curious about your opinion, whether these features could be useful or not.

**Comment by sacdevops** [comment]
I had the last month many situation where unit sized like this would help or make it just easier as the current solution.
Would be a really cool feature and would open new ways to place any controls in frontend.
Imagine a stacklayout or image that takes just 70% of width e.g. instead of using grid with rowdefinitions 7* and 3* or to calculate even the sizes and bind them.

**Reply by zxopink** [reply]
Been on the look for ways to use relative units, it would make things much easier than messing around with grids or hard-coding element size/position
